packages/atomic/src/components/search-box-suggestions/atomic-search-box-instant-results/atomic-search-box-instant-results.tsx

```diff
@@ -0,0 +1,97 @@
+import {Component, Element, State, h} from '@stencil/core';
+import {buildInstantResults, InstantResults, Result} from '@coveo/headless';
+
+import {
+  dispatchSearchBoxSuggestionsEvent,
+  SearchBoxSuggestionItem,
+  SearchBoxSuggestions,
+  SearchBoxSuggestionsBindings,
+} from '../suggestions-common';
+
+/**
+ * The `atomic-search-box-instant-results` component can be added as a child of an `atomic-search-box` component, allowing for the configuration of instant results behavior.
+ * @internal
+ */
+@Component({
+  tag: 'atomic-search-box-instant-results',
+  shadow: true,
+})
+export class AtomicSearchBoxRecentQueries {
+  private bindings!: SearchBoxSuggestionsBindings;
+
+  @Element() private host!: HTMLElement;
+
+  @State() public error!: Error;
+  private instantResults!: InstantResults;
+
+  private results: Result[] = [];
+
+  componentWillLoad() {
+    try {
+      dispatchSearchBoxSuggestionsEvent((bindings) => {
+        this.bindings = bindings;
+        return this.initialize();
+      }, this.host);
+    } catch (error) {
+      this.error = error as Error;
+    }
+  }
+
+  private renderItems(): SearchBoxSuggestionItem[] {
+    const results = this.instantResults.state.results.length
+      ? this.instantResults.state.results
+      : this.results;
+    return results.map((result, i) => ({
+      key: `instant-result-${i}`,
+      query: '',
+      content: <div class="flex items-center break-all">{result.title}</div>,
+      onSelect: () => {
+        // TODO: ADD LOGS?
```

Comment:
I think re-using `LinkWithResultAnalytics` here  (refactored if needed)  should do the trick... 🤔 Totally fine to leave it for another PR !